client/apps/game/src/ui/components/military/army-management-card.tsx

```diff
@@ -76,10 +75,25 @@ export const ArmyCreate = ({ owner_entity, army, armyManager, isExplorer, guardS
   const handleBuyArmy = async (isExplorer: boolean, troopType: TroopType, troopTier: TroopTier, troopCount: number) => {
     setIsLoading(true);
 
+    const homeDirection =
```

Comment:
my point of view is that if something only needs to be used on a click , then compute that thing when user clicks, instead of having it computed every time